assets/js/src/newsletters/send.jsx

```diff
@@ -74,6 +74,38 @@ const NewsletterSend = createReactClass({ // eslint-disable-line react/prefer-es
     return jQuery('#mailpoet_newsletter').parsley().isValid();
   },
 
+  isValidFromAddress: function isValidFromAddress() {
+    const fromAddress = jQuery('#field_sender_address').val();
+    if (window.mailpoet_mta_method !== 'MailPoet') {
+      return true;
+    }
+    return this.state.authorizedEmailAddresses.indexOf(fromAddress) !== -1;
+  },
+
+  showInvalidFromAddressError: function showInvalidFromAddressError() {
+    const errorMessage = MailPoet.I18n.t('newsletterInvalidFromAddress')
+      .replace('%$1s', jQuery('#field_sender_address').val())
+      .replace('[i]', '<i>')
```

Comment:
Didn't know about using simple tags like `<i>` in strings 👍 

About the `ReactStringReplace` - I was thinking about using it, but ended up with `.replace` because it looks more readable (especially now with removed replace for `[i]`).

```
const errorMessage = ReactStringReplace(
  MailPoet.I18n.t('newsletterInvalidFromAddress')
    .replace('%$1s', this.state.item.sender_address),
  /\[link\](.*?)\[\/link\]/g,
  (match) => (
    <a href="https://account.mailpoet.com/authorization" target="_blank" rel="noopener noreferrer">{match}</a>
  )
);
```

vs.

```
const errorMessage = MailPoet.I18n.t('newsletterInvalidFromAddress')
  .replace('%$1s', this.state.item.sender_address)
  .replace('[link]', '<a href="https://account.mailpoet.com/authorization" target="_blank" rel="noopener noreferrer">')
  .replace('[/link]', '</a>');
```

What do you think?